dts/bindings/serial/nuvoton,numicro-uart.yaml

```diff
@@ -2,8 +2,11 @@ description: NUVOTON NUMICRO FAMILY UART
 
 compatible: "nuvoton,numicro-uart"
 
-include: uart-controller.yaml
+include: [uart-controller.yaml, pinctrl-device.yaml]
 
 properties:
     reg:
       required: true
+
+    pinctrl-0:
+      required: true
```

Comment:
see my previous comment: you are forcing pinctrl, so driver doesn't need to make its use conditional.